Author: aszlig <aszlig@nix.build>
Date:   Tue Mar 27 07:32:57 2018 +0200

    Do not restrict email domains
    
    Those restrictions are on habitica.com and habitrpg.com, so they don't
    apply to a standalone setup anyway.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: dont-restrict-email-domains.patch

diff --git a/test/api/v3/integration/user/auth/POST-register_local.test.js b/test/api/v3/integration/user/auth/POST-register_local.test.js
index 6bc46c2d5..bd108aa72 100644
--- a/test/api/v3/integration/user/auth/POST-register_local.test.js
+++ b/test/api/v3/integration/user/auth/POST-register_local.test.js
@@ -372,40 +372,6 @@ describe('POST /user/auth/local/register', () => {
       expect(user.auth.local.email).to.equal(email.toLowerCase());
     });
 
-    it('fails on a habitica.com email', async () => {
-      let username = generateRandomUserName();
-      let email = `${username}@habitica.com`;
-      let password = 'password';
-
-      await expect(api.post('/user/auth/local/register', {
-        username,
-        email,
-        password,
-        confirmPassword: password,
-      })).to.eventually.be.rejected.and.eql({
-        code: 400,
-        error: 'BadRequest',
-        message: 'User validation failed',
-      });
-    });
-
-    it('fails on a habitrpg.com email', async () => {
-      let username = generateRandomUserName();
-      let email = `${username}@habitrpg.com`;
-      let password = 'password';
-
-      await expect(api.post('/user/auth/local/register', {
-        username,
-        email,
-        password,
-        confirmPassword: password,
-      })).to.eventually.be.rejected.and.eql({
-        code: 400,
-        error: 'BadRequest',
-        message: 'User validation failed',
-      });
-    });
-
     it('requires a password', async () => {
       let username = generateRandomUserName();
       let email = `${username}@example.com`;
diff --git a/website/server/models/user/schema.js b/website/server/models/user/schema.js
index 7764c35d2..e951de610 100644
--- a/website/server/models/user/schema.js
+++ b/website/server/models/user/schema.js
@@ -14,8 +14,6 @@ import {
 
 const Schema = mongoose.Schema;
 
-const RESTRICTED_EMAIL_DOMAINS = Object.freeze(['habitica.com', 'habitrpg.com']);
-
 // User schema definition
 let schema = new Schema({
   apiToken: {
@@ -31,15 +29,6 @@ let schema = new Schema({
         validate: [{
           validator: validator.isEmail,
           message: shared.i18n.t('invalidEmail'),
-        }, {
-          validator (email) {
-            let lowercaseEmail = email.toLowerCase();
-
-            return RESTRICTED_EMAIL_DOMAINS.every((domain) => {
-              return !lowercaseEmail.endsWith(`@${domain}`);
-            });
-          },
-          message: shared.i18n.t('invalidEmailDomain', { domains: RESTRICTED_EMAIL_DOMAINS.join(', ')}),
         }],
       },
       username: {
